code/datums/emotes.dm

```diff
@@ -238,6 +241,11 @@
 		return FALSE
 	if(is_type_in_typecache(user, mob_type_blacklist_typecache))
 		return FALSE
+	if(istype(user, /mob/living/carbon/human))
+		var/mob/living/carbon/human/H = user
+		if(species_type_whitelist_typecache && H.dna && !is_type_in_typecache(H.dna.species, species_type_whitelist_typecache))
```

Comment:
```suggestion
		if(species_type_whitelist_typecache && H.dna && !species_type_whitelist_typecache[H.dna.species])
```
is_type_in_typecache() is pointless. shitty macro. this applies to all uses of it.